builtin/logical/pki/path_tidy.go

```diff
@@ -872,10 +898,31 @@ func (b *backend) doTidyCertStore(ctx context.Context, req *logical.Request, log
 
 		cert, err := x509.ParseCertificate(certEntry.Value)
 		if err != nil {
-			return fmt.Errorf("unable to parse stored certificate with serial %q: %w", serial, err)
+
+			// only log warning once
+			if !haveWarned {
+				logger.Warn("unable to parse stored certificate with serial %q: %w; %s. Other invalid certificates may exist.", serial, err,
+					func() string {
+						if config.InvalidCerts {
+							return "tidying up since it is not usable"
+						}
+						return "tidy by enabling tidy_invalid_certs"
+					}())
```

Comment:
Ahh, creative! Just a note though, this isn't a `printf`-style logger, this is a `key=value` style logger with explicit message.

So I think we'd need something like the following:

```suggestion
				msg := "Unable to parse stored certificate. Other invalid certificates may exist; "
				if config.InvalidCerts {
					msg += "tidying up since it is not usable."
				} else {
					msg += "tidy by enabling tidy_invalid_certs=true."
				}
				logger.Warn(msg, "serial", serial, "err", err)
```

as otherwise you'll get weird logger messages. I believe this wouldn't err, but would output something like:

```
[WARN] msg="unable to parse stored certificate with serial %q: %w; %s. Other invalid certificates may exist."  1d:de:f6:bc:77:2e:56:cb:b7:00:bc:5b:ea:db:37:63:0a:3e:2d:98="x509: certificate contains duplicate extensions"
```

which wouldn't make as much sense.

Reply:
Ah interesting, I will keep this in mind for future logging :D Thank you